### DISCUSSION TITLE
Difference in output when running FastAPI endpoint vs standalone Python script for a document retrieval system using ChromaDB, LangChain, OpenAI

### DISCUSSION THREAD

**Discussion by burberg92** [fastapi/fastapi#10251] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
Processing.py code :

# Initialization

# Imports and environment setup
from langchain.document_loaders import PyPDFLoader
from langchain.text_splitter import RecursiveCharacterTextSplitter
from langchain.embeddings.openai import OpenAIEmbeddings
from langchain.vectorstores import Chroma
from langchain.chat_models import ChatOpenAI
from langchain.chains import RetrievalQA
from langchain.embeddings.openai import OpenAIEmbeddings

import os

# Environment Variables
os.environ["OPENAI_API_KEY"] = ""

embeddings = OpenAIEmbeddings()

# Load pre-embedded vector database (assuming Chroma provides a load function)

vectordb = Chroma(persist_directory="/Users/rodrigoburberg/Desktop/RAG_MPEP/RAG_MPEP/assets", embedding_function=embeddings)

# Set up the chat model
llm = ChatOpenAI(temperature=0)
doc_retriever = vectordb.as_retriever()
mpep100_qa = RetrievalQA.from_chain_type(llm=llm, chain_type="stuff", retriever=doc_retriever)


def query_document(user_query):
    result = mpep100_qa.run(user_query)
    return result

print(query_document('what are Content of the Statements in patents'))



main.py code:

from fastapi import FastAPI, HTTPException
from utils.processing import query_document as process_query
from fastapi.middleware.cors import CORSMiddleware
from pydantic import BaseModel


class Input(BaseModel):
    human_input: str 

class Output(BaseModel):
    output: str 


Input.update_forward_refs()
Output.update_forward_refs()

app = FastAPI()

@app.post("/query")
async def query_endpoint(query: Input):
    try:
        response = process_query(query.human_input)
        output = Output(output =response)
        return output
    except:
        raise HTTPException(status_code=500, detail="An error occurred while processing the query.")
```

### Description

I am developing a document retrieval system RAG using the langchain library. When I run my Python script (processing.py) directly, it retrieves a contextually accurate response from a given document. However, when I integrated the same logic into a FastAPI endpoint (main.py), the output from the endpoint lacks the context found in the standalone script's output.

The same input was used - 'what are Content of the Statements in patents'

output from processing.py:

The content of the statements in patents refers to the information that must be included in a statement under oath filed with a patent application. According to the provided context, the statement should set forth "the full facts" surrounding the conception and making of the invention or discovery. This includes information about any contracts, subcontracts, or arrangements with or for the benefit of the Atomic Energy Commission or the National Aeronautics and Space Administration. If the applicant does not have firsthand knowledge of these details, they should identify the source of their information. Additionally, if there are any changes in inventorship during the application process, a supplemental statement executed by the added inventor(s) is required.

output from main.py (testing swagger ui):

{ "output": "The content of statements in patents typically includes a detailed description of the invention, including its technical features, functionality, and any unique aspects. It may also include claims, which define the scope of protection sought for the invention. Additionally, patents often include background information, prior art references, drawings or diagrams, and any supporting data or examples." }

### Operating System

macOS

### FastAPI Version

0.103.1

### Pydantic Version

1.10.8

### Python Version

3.8.18

**Comment by Nikhil-218** [comment]
@burberg92 Hello, 
I'm also facing a similar issue. Did you find any solution, or what is the root cause of this?
